spec/openapi.yaml

```diff
@@ -53,182 +156,498 @@ paths:
+        number_of_pages:
+          type: integer
+        page:
           type: integer
-    inline_response_200:
-      type: object
-      properties:
         records_per_page:
           type: integer
-        apiVersion:
+        status:
           type: string
         status_code:
           type: integer
+        total_count:
+          type: integer
+    Language_data:
+      type: object
+      properties:
+        id:
+          type: integer
+        name:
+          type: string
+    
+    SearchResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
         data:
           type: array
           items:
-            $ref: '#/components/schemas/inline_response_200_data'
+            type: object
+            $ref: '#/components/schemas/Resource_data'
         number_of_pages:
           type: integer
+        page:
+          type: integer
+        records_per_page:
+          type: integer
+        status:
+          type: string
+        status_code:
+          type: integer
         total_count:
           type: integer
+    FilteredResourcesResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+        data:
+          type: array
+          items:
+            type: object
+            $ref: '#/components/schemas/Resource_data'
         has_next:
-          type: boolean
-        has_prev:
-          type: boolean
+          type: integer
+        has_previous:
+          type: integer
+        number_of_pages:
+          type: integer
         page:
           type: integer
+        records_per_page:
+          type: integer
         status:
           type: string
+        status_code:
+          type: integer
+        total_count:
+          type: integer
+    ApiKeyResponse:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+        data:
+          $ref: '#/components/schemas/ApiKeyResponse_data'
+        status:
+          type: string
+        status_code:
+          type: integer
+    ApiKeyRequest:
+      type: object
+      $ref: '#/components/schemas/ApiKeyRequest_data'
     Resource_data:
       type: object
       properties:
-        last_updated:
+        category:
           type: string
-        notes:
+        created_at:
           type: string
-        times_clicked:
+          format: date-time
+        downvotes:
           type: integer
+        id:
+          type: integer
+        last_updated:
+          type: string
+          format: date-time
         languages:
           type: array
           items:
             type: string
-        upvotes:
-          type: integer
         name:
           type: string
+        notes:
+          type: string
         paid:
           type: boolean
-        created_at:
-          type: string
-        id:
+        times_clicked:
           type: integer
-        category:
-          type: string
-        downvotes:
+        upvotes:
```

Comment:
When you're creating a resource, you should not have `upvotes`. The API shouldn't accept this anyway, but the docs need to reflect that this is not a valid parameter. Remember that the Request and the Response are not necessarily 1 to 1.

This endpoint should not accept any of the following:
`created_at`
`downvotes`
`id`
`last_updated`
`times_clicked`
`upvotes`

It should accept:
`category`
`languages`
`name`
`notes`
`paid`
`url`